transport/tchannel/outbound.go

```diff
@@ -184,13 +188,28 @@ func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Pe
 		ApplicationError: res.ApplicationError(),
 		ApplicationErrorMeta: &transport.ApplicationErrorMeta{
 			Err:  nil,
-			Name: "",
-			Code: nil,
+			Name: applicationErrorName,
+			Code: applicationErrorCode,
 		},
 	}
 	return resp, err
 }
 
+func getResponseErrorCodeFromHeaders(headers transport.Headers) *yarpcerrors.Code {
```

Comment:
nit: let's clarify that this is for the application error code (since there's existing logic around retrieving regular error codes)
```suggestion
func getApplicationErrorCodeFromHeaders(headers transport.Headers) *yarpcerrors.Code {
```